dashboard/src/shared/Auth.ts

```diff
@@ -65,21 +68,23 @@ export class Auth {
     }
   }
 
-  // fetchOIDCToken does a HEAD request to collect the Bearer token
-  // from the authorization header if exists
-  public static async fetchOIDCToken(): Promise<string | null> {
+  // isAuthenticatedWithCookie() does a HEAD request (without any token obviously)
+  // to determine if the request is authenticated (ie. not a 401). Unfortunately
+  // Kubernetes defaulting to allow anonymous requests means that this will be a 403
+  // even if there are no credentials, so we additionally check the message and assume
+  // we are authenticated with a cookie if a 403 is not for an anon user.
+  public static async isAuthenticatedWithCookie(): Promise<boolean> {
     try {
-      const { headers } = await Axios.head("");
-      if (headers && headers.authorization) {
-        const tokenMatch = (headers.authorization as string).match(/Bearer\s(.*)/);
-        if (tokenMatch) {
-          return tokenMatch[1];
-        }
-      }
+      await Axios.get(APIBase + "/");
```

Comment:
this should be `Axios.head`, no? (or maybe the comment is wrong)

Reply:
The comment is wrong (need to check the body of the error response), but already fixed in the following branch.